smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/ServerGenerator.java

```diff
@@ -0,0 +1,152 @@
+            writer.openBlock("private serdeContextBase = {", "};", () -> {
+                writer.write("base64Encoder: toBase64,");
+                writer.write("base64Decoder: fromBase64,");
+                writer.write("utf8Encoder: toUtf8,");
+                writer.write("utf8Decoder: fromUtf8,");
+                writer.write("streamCollector: streamCollector,");
+                writer.write("requestHandler: new NodeHttpHandler(),");
+                writer.write("disableHostPrefix: true");
+            });
+            writer.write("/**");
+            writer.write(" * Construct a $LService handler.", serviceName);
+            writer.write(" * @param service The {@link $LService} implementation that supplies", serviceName);
+            writer.write(" *                the business logic for $LService", serviceName);
+            writer.write(" * @param mux The {@link Mux} that determines which service and operation are being");
+            writer.write(" *            invoked by a given {@link HttpRequest}");
+            writer.write(" * @param serializers An {@link OperationSerializer} for each operation in");
+            writer.write(" *                    $LService that handles deserialization of requests and", serviceName);
+            writer.write(" *                    serialization of responses");
+            writer.write(" */");
+            writer.openBlock("constructor(service: $1LService, "
+                            + "mux: Mux<$1S, $2L>, "
+                            + "serializers: Record<$2L, OperationSerializer<$1LService[$2L]>>) {", "}",
+                    serviceName, operationsTypeName, () -> {
+                writer.write("this.service = service;");
+                writer.write("this.mux = mux;");
+                writer.write("this.serializers = serializers;");
+            });
+            writer.openBlock("async handle(request: HttpRequest): Promise<HttpResponse> {", "}", () -> {
+                writer.write("const target = this.mux.match(request);");
+                writer.openBlock("if (target === undefined) {", "}", () -> {
+                    writer.write("throw new Error(`Could not match any operation to $${request.method} "
+                            + "$${request.path} $${JSON.stringify(request.query)}`);");
+                });
+                writer.openBlock("switch (target.operation) {", "}", () -> {
+                    for (OperationShape operation : operations) {
+                        generateHandlerCase(writer, operation);
+                    }
+                });
+            });
+        });
+    }
+
+    private static void generateHandlerCase(TypeScriptWriter writer, OperationShape operation) {
+        String opName = operation.getId().getName();
+        writer.openBlock("case $S :", "", opName, () -> {
+            writer.openBlock("let input = await this.serializers.$L.deserialize(request, {", "});", opName, () -> {
```

Comment:
The output type of this is the union type and so it can't be passed into the service interface that expects a singular concrete type. You'll need to either `any` this or append `as ConcreteInput` or something.

Reply:
I was wondering when having a model with just one test operation was going to bite me.

I think this actually points at the typing of the `Record` and `OperationSerializer` being not quite right. I am against `any`ing anything, I think I have a safe, but ugly temporary stopgap.

Reply:
Yeah I think the only other real option would be to abandon `Record`. Cool as it is, it's way more useful when you're not generating the code. A static mapping might ultimately be necessary, but that can be done in a followup.